pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -120,6 +127,10 @@ def __init__(
             )
         self.monitor = monitor or "early_stop_on"
 
+    @property
+    def state_key(self) -> str:
+        return self._generate_state_key(monitor=self.monitor, mode=self.mode)
```

Comment:
noob question: Could we automate this using save_hyperparameters(self) and use `self.hparams` to generate the state key ?

Reply:
I think I answered this question already: https://github.com/PyTorchLightning/pytorch-lightning/pull/7187#discussion_r678957513

In summary, auto-generating a key from all init args will be too restrictive for users.